### PR TITLE
New method to check service availability

### PR DESCRIPTION
**Pull request by leoporoli**
It

### CODE REVIEW

File: core-api/api_container_service.proto
Comment (mieubrisse) [inline core-api/api_container_service.proto:29]: Can we call this `WaitForEndpointAvailability`, to more accurately reflect its purpose?

File: core-api/api_container_service.proto
Comment (mieubrisse) [inline core-api/api_container_service.proto:175]: Same rename here

File: core-api/api_container_service.proto
Comment (mieubrisse) [inline core-api/api_container_service.proto:178]: Can we add explanatory comments to each variable here, to make them more self-documenting?

File: docs/changelog.md
Comment (mieubrisse) [inline docs/changelog.md:3]: s/CheckServiceAvailability/WaitForEndpointAvailability/

Comment (mieubrisse) [reply docs/changelog.md:3]: Recommended rephase: "Added a new method `WaitForEndpointAvailability` in NetworkContext that can be used to wait until a service's endpoint becomes available"

File: core-api/api_container_service.proto
Comment (mieubrisse) [inline core-api/api_container_service.proto:179]: Also recommend using `uint` rathe rthan `int` for these variables, since it doesn't make sense for them to be negative

File: golang/networks/network_context.go
Comment (mieubrisse) [inline golang/networks/network_context.go:269]: Can we add the same `Docs available at....` comment that exists on all the other funcs?

File: core-api/api_container_service.proto
Comment (leoporoli) [reply core-api/api_container_service.proto:29]: Yes! I'll do it

File: core-api/api_container_service.proto
Comment (leoporoli) [reply core-api/api_container_service.proto:179]: Great suggest!

**Comment by leoporoli** [comment]
@mieubrisse all comments resolved, waiting for a second review.

File: core-api/api_container_service.proto
Comment (mieubrisse) [inline core-api/api_container_service.proto:182]: s/executes/executing/

File: core-api/api_container_service.proto
Comment (mieubrisse) [inline core-api/api_container_service.proto:184]: s/until it gets a success response from the service/until giving up and returning an error/

File: core-api/api_container_service.proto
Comment (mieubrisse) [inline core-api/api_container_service.proto:188]: Recommend instead "If the endpoint returns this value, the service will be marked as available (e.g. `Hello World`)"

File: docs/changelog.md
Comment (mieubrisse) [inline docs/changelog.md:2]: Needs deleting

File: golang/networks/network_context.go
Comment (mieubrisse) [reply golang/networks/network_context.go:270]: Let's make this name match the `WaitForEndpointAvailability`

File: core-api/api_container_service.proto
Comment (mieubrisse) [inline core-api/api_container_service.proto:179]: shouldn't this be a `uint`?

**Review by mieubrisse — APPROVED** [review]
Approved pending comments

**Pull request merged**
